### ISSUE TITLE
ICE "break to unknown label" when while has a labeled break of itself in condition

### ISSUE THREAD

**Issue by leodasvacas** [rust-lang/rust#37353]
```rust
fn main() {
    loop { 'w: while break 'w { } }
}
```
ICEs with:
```
rustc 1.14.0-nightly (6e8f92f11 2016-10-07)
error: internal compiler error: ../src/librustc/middle/liveness.rs:1057: break to unknown label
 --> <anon>:3:19
  |
3 |         'w: while break 'w { }
  |                   ^^^^^^^^
```
Code like this was found in issue #21751.

**Comment by jonas-schievink** [comment]
This complains about `break outside of loop`:

```rust
fn main() {
    while break {}
}
```

While this loops forever:
```rust
fn main() {
    loop { while break {} }
}
```

This means that the `break` doesn't actually break out of the outer loop, as the first error might suggest.

**Comment by TimNN** [comment]
Regression from 1.11.0 to 1.12.0.

On nightly, this was introduced between `nightly-2016-07-15` and `nightly-2016-07-16` ([Changes](https://github.com/rust-lang/rust/compare/b6c1ef374...4db1874f4)).

Introduced in #34570, cc @jseyfried.

**Comment by jseyfried** [comment]
Fixed in #37360.
